apis/python/setup.cfg

```diff
@@ -3,12 +3,6 @@ profile = black
 known_third_party = anndata
 skip = dist_links
 
-[flake8]
```

Comment:
Because [ruff is configurable only via pyproject.toml and the command line](https://github.com/charliermarsh/ruff#configuration), it doesn't understand setup.cfg. 

By the way, `isort` and `mypy` are configurable via either pyproject.toml or setup.cfg (or other config files). In fact only flake8 is the odd one out that stubbornly refuses to support pyproject.toml. So this PR opens the way to move the remaining setup.cfg contents to pyproject.toml and have one less file file to maintain.